controllers/operandrequest/reconcile_operand.go

```diff
@@ -948,16 +948,48 @@ func (r *Reconciler) createK8sResource(ctx context.Context, k8sResTemplate unstr
 	name := k8sResTemplate.GetName()
 	namespace := k8sResTemplate.GetNamespace()
 
+	if kind == "Route" {
+		host, _, err := unstructured.NestedString(k8sResTemplate.Object, "spec", "host")
+		klog.Info("11111 host %v: ", host)
+
+		if err != nil {
+			klog.Errorf("failed to get spec.host for Route: %v", err)
+			return err
+		}
+
+		// Calculate and set hash for spec.host
+		hostHash := util.CalculateHash(host)
+		if newAnnotations == nil {
+			newAnnotations = make(map[string]string)
+		}
+		newAnnotations["openshift.io/keycloak-host.hashedData"] = hostHash
+		// klog.Infof("22222 Set host hash annotation: %s", hostHash)
+	}
+
 	if k8sResConfig != nil {
 		k8sResConfigDecoded := make(map[string]interface{})
 		k8sResConfigUnmarshalErr := json.Unmarshal(k8sResConfig.Raw, &k8sResConfigDecoded)
 		if k8sResConfigUnmarshalErr != nil {
 			klog.Errorf("failed to unmarshal k8s Resource Config: %v", k8sResConfigUnmarshalErr)
 		}
-
+		klog.Infof("11111 k8sResConfig: %v", k8sResConfigDecoded)
 		for k, v := range k8sResConfigDecoded {
 			k8sResTemplate.Object[k] = v
 		}
+
+		if kind == "Route" {
+			if host, found := k8sResConfigDecoded["spec"].(map[string]interface{})["host"].(string); found {
+				hostHash := util.CalculateHash(host)
+
+				if newAnnotations == nil {
+					newAnnotations = make(map[string]string)
+				}
+				newAnnotations["openshift.io/keycloak-host.hashedData"] = hostHash
```

Comment:
can we use odlm specific label? For example `operator.ibm.com/odlm.route.hashedData`